src/hosted-buttons/types.js

```diff
@@ -43,7 +43,14 @@ export type HostedButtonDetailsParams =
       shape: string,
       color: string,
       label: string,
+      height: number,
     |},
+    version: string,
+    preferences: {|
+      secondButton: "recommended" | "venmo" | "paylater" | "none",
```

Comment:
instead of going with object, can we keep this as an Array instead?

Reply:
Do you mean for `preferences`? 🤔

Reply:
Yes. I am thinking this can be expanded to include an ordered preference of buttons

Reply:
Ahh that would be interesting.  We'd probably want to take that up with the NCP team. Right now,  we are getting preferences from `response.body.button_details.preferences` & that's structured like this:
```
preferences: {
    second_button: "venmo", // Merchant's preferred second button if eligible
    eligible_funding_methods: ["venmo", "paylater"], // Ordered list of the buttons the merchant is eligible for
}
```

Are you saying that we should put some logic here to combine these into an array of ordered preference of eligible buttons?  I'm totally down for this, I just might change the name of `preferences` so that it doesn't get mixed up with the `preferences` object in the /form-fields response.

Reply:
We can maybe call it `filteredPreferences` or something. In this case it would be `["venmo", "paylater"]`.

But if `second_button = "paylater"`, it would be `["paylater", "venmo"]`. 

Or if venmo wasn't eligible, it would just be `["paylater"]`.

Is that what you had in mind?

Reply:
because we'll still have to do our own `button.isEligible()` checks before we render any of these